rend3-anim/src/lib.rs

```diff
@@ -213,6 +214,38 @@ pub fn pose_animation_frame(
     let animation = &scene.animations[animation_index];
     let time = time.clamp(0.0, animation.inner.duration);
 
+
+    for (&node_idx, channels) in &animation.inner.channels {
+
+        let local_transform = instance.nodes[node_idx].inner.local_transform;
+        let (bind_scale, bind_rotation, bind_translation) = local_transform.to_scale_rotation_translation();
+
+        let translation = channels
+            .translation
+            .as_ref()
+            .map(|tra| sample_at_time(tra, time))
+            .unwrap_or(bind_translation);
+        let rotation = channels
+            .rotation
+            .as_ref()
+            .map(|rot| sample_at_time(rot, time))
+            .unwrap_or(bind_rotation);
+        let scale = channels
+            .scale
+            .as_ref()
+            .map(|sca| sample_at_time(sca, time))
+            .unwrap_or(bind_scale);
+
+        let matrix = Mat4::from_scale_rotation_translation(scale, rotation, translation);
+
+        if let Some(object) = instance.nodes[node_idx].inner.object.borrow() {
```

Comment:
Manually using the `Borrow` trait here feels a bit odd. You could solve this in three different ways:

*Using borrow syntax instead (equivalent to .borrow())*
```rust
if let Some(object) = &instance.nodes[node_idx].inner.object {
```

*Using `Option::as_ref()`*
```rust
if let Some(object) = instance.nodes[node_idx].inner.object.as_ref() {
```

*Using the `ref` keyword inside your if let pattern*
```rust
if let Some(ref object) = instance.nodes[node_idx].inner.object {
```

All three should be equivalent, but personally I prefer the latter as being more explicit about what's going on.

Reply:
Cool, yeah - I'm a Rust noob and thus not very critical with regards to IDE suggestions. Will change this.

Reply:
Regarding the animation example - Good idea, I'll try it out.